stack.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"time"
+
+	"errors"
+
+	"github.com/distributeddesigns/currency"
+)
+
+type buyItem struct {
+	amount         currency.Currency
+	numStocks      uint
+	price          currency.Currency
+	stock          string
+	quoteTimeStamp time.Time
+}
+
+type buyStack []buyItem
+
+func (s buyStack) isEmpty() bool {
+	return len(s) == 0
+}
+
+func (bItem buyItem) isExpired() bool {
+	return bItem.quoteTimeStamp.Before(time.Now().Add(time.Second * -10))
```

Comment:
Needs to be add.

From the golang docs at https://golang.org/pkg/time/#Time.Sub

func (Time) Sub

func (t Time) Sub(u Time) Duration
Sub returns the duration t-u. If the result exceeds the maximum (or minimum) value that can be stored in a Duration, the maximum (or minimum) duration will be returned. To compute t-d for a duration d, use t.Add(-d).